### PR TITLE
chore(datastore): refactor integration tests with v2 schema models

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:*

*Description of changes:*

This PR is implemented based on the release-candidate branch.

* Refactored DataStore integration tests with GraphQL transformer v2 schema
* Reduced number of test models for future API_SYNC enabled integration tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cshfang — APPROVED** [review]
Think I mostly only saw typos and had a small question. Otherwise lgtm.

File: packages/amplify_datastore/example/integration_test/model_type_test.dart [hunk 1]
```diff
@@ -31,96 +31,106 @@ void main() {
...
+      var models = List.generate(
+          5, (_) => ModelWithAppsyncScalarTypes(listOfStringValue: list));
       testModelOperations(models: models);
     });
 
     group('List<String> (with null value)', () {
-      var models = List.generate(5, (_) => StringListTypeModel());
+      var models = List.generate(5, (_) => ModelWithAppsyncScalarTypes());
       testModelOperations(models: models);
     });
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply packages/amplify_datastore/example/integration_test/model_type_test.dart:55]: Ahh yes, I think so, as we are using the same model to perform all tests here.

File: packages/amplify_datastore/example/integration_test/query_test/query_predicate_test/query_predicate_test.dart [hunk 2]
```diff
@@ -18,20 +18,30 @@ import 'package:flutter_test/flutter_test.dart';
 
 import 'string_query_predicate_test.dart' as string_query_predicate_tests;
 import 'int_query_predicate_test.dart' as int_query_predicate_tests;
-import 'double_query_predicate_test.dart' as double_query_predicate_tests;
-import 'bool_query_predicate_test.dart' as bool_query_predicate_tests;
+import 'float_query_predicate_test.dart' as float_query_predicate_tests;
+import 'boolean_query_predicate_test.dart' as boolean_query_predicate_tests;
 import 'enum_query_predicate_test.dart' as enum_query_predicate_tests;
 import 'compound_query_predicate_test.dart' as compound_query_predicate_tests;
+import 'aws_date_query_predicate_test.dart' as aws_date_query_predicate_test;
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/integration_test/query_test/query_predicate_test/query_predicate_test.dart:25]: Huh. Were these omitted originally? Good catch.

**Review by Jordan-Nelson — APPROVED** [review]
LGTM. Thanks for refactoring this

**Pull request merged**